modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/transaction/DistributedTransaction.java

```diff
@@ -60,18 +76,61 @@ public boolean isTransactional() {
         return false;
     }
 
-    @Override public void onClose() {
-
+    @Override
+    public void onClose() {
     }
 
     @Override
-    public void start(Xid xid, boolean join, boolean resume) {
-
+    public void start(Xid xid, int sessionId, boolean join, boolean resume) throws ValidationException {
+        if (join && resume) {
+            throw new ValidationException("Cannot start a branch with both join and resume set " + xid);
+        }
+
+        Branch branch = transactionRegistry.getBranch(xid);
+        if (join) {
+            if (Objects.isNull(branch)) {
+                throw new ValidationException(SAME_XID_ERROR_MSG + xid);
+            }
+            branch.associateSession(sessionId);
+        } else if (resume) {
+            if (Objects.isNull(branch)) {
+                throw new ValidationException(SAME_XID_ERROR_MSG + xid);
+            }
+            branch.resumeSession(sessionId);
+        } else {
+            if (Objects.nonNull(branch)) {
+                throw new ValidationException("Xid " + xid + " cannot be started as it is already known");
+            }
+            branch = branchFactory.createBranch(xid);
+            transactionRegistry.register(branch);
+            branch.associateSession(sessionId);
```

Comment:
Will it make sense to move line 105 and 106 to the factory class since we need to do that every time we create a branch?

Reply:
then it would be createAndRegister operation. Another option is to have a transactionRegistry#createBranch where we create the branch and register. Btw this is the only place we create a branch ATM for both local and distributed transactions.

Reply:
Anyway I'll update this as needed when I'm gonna implement retrieving stored prepared transactions from database. In that case we will have to create branches as well.